small-talk/db/registerToDB.js

```diff
@@ -5,18 +5,29 @@ import bcrypt from "bcryptjs";
 export default async function handleRegister(username, password, firstname, lastname, dob){ // function that sends the credentials to the database
   const mongoboi = new Mongoboi(uri, "Users");
   const date = new Date(dob).getTime() / 1000; // return the inputted date as a Unix timestamp
-  await mongoboi.connect();
-
-  const hashedpass = await bcrypt.hash(password, 10);
-
+  //await mongoboi.connect();
+  
   const newUser = { // schema for holding the values into certain fields for database organization
     username: username,
-    password: hashedpass,
     firstname: firstname,
     lastname: lastname,
     dob: date,
   };
 
+  bcrypt.genSalt(10, async function(saltError, salt){ // generate a salt for password
+    if (saltError) {
+      return saltError
+    }
+    else {
+      bcrypt.hash(password, salt, function (hashError, hash){ // use hash algorithm to generate hashed password w/ salt
+        if (hashError){
+          return hashError
+        }
+        newUser.password = hash // if checks pass, implement hash/salt password to schema
+      })
+    }
+  })
+
```

Comment:
Very cool how the salt is simply stored with the hash, so there's no extra step in checking when they go to log in again.